tls/s2n_psk.h

```diff
@@ -89,9 +69,24 @@ int s2n_psk_calculate_binder(struct s2n_psk *psk, const struct s2n_blob *binder_
 int s2n_psk_verify_binder(struct s2n_connection *conn, struct s2n_psk *psk,
         const struct s2n_blob *partial_client_hello, struct s2n_blob *binder_to_verify);
 
-typedef int (*s2n_psk_selection_callback)(struct s2n_connection *conn, 
+typedef int (*s2n_psk_selection_callback)(struct s2n_connection *conn,
                                           struct s2n_psk_identity *identities, size_t identities_length,
-                                          uint16_t *chosen_wire_index);                     
+                                          uint16_t *chosen_wire_index);
 /* This function will be labeled S2N_API and become a publicly visible api once we release the psk API. */
 int s2n_config_set_psk_selection_callback(struct s2n_connection *conn, s2n_psk_selection_callback cb);
 
+/* Public Interface -- will be made visible and moved to s2n.h when the PSK feature is released */
+
+struct s2n_psk;
+typedef enum {
+    S2N_PSK_HMAC_SHA224 = 0,
+    S2N_PSK_HMAC_SHA256,
+    S2N_PSK_HMAC_SHA384,
+} s2n_psk_hmac;
+
+struct s2n_psk* s2n_external_psk_new();
```

Comment:
Do we want a unique type for external/internal PSKs? If there are some exclusive features to either one it might be a good idea in order to prevent confusion about what I can/can't do with a `s2n_psk`.

Reply:
If customers are interacting with a PSK, it's an external PSK. Resumption PSKs are handled via tickets.